pgmpy/inference/mplp.py

```diff
@@ -0,0 +1,752 @@
+import itertools
+import numpy as np
+from operator import sub
+from operator import add
+import queue
+import math
+from collections import namedtuple
+from pgmpy.models import MarkovModel
+from pgmpy.inference import ApproxInference
+
+
+class MPLP(ApproxInference):
+    """
+    Class for performing inference using Max-Product Linear Programming Relaxations.
+    We use the Max-Product Linear Programming (MPLP) algorithm to minimize the dual .
+    """
+
+    def __init__(self, model):
+        super().__init__(model)
+
+        if not isinstance(model, MarkovModel):
```

Comment:
@kislayabhi First check for the model and then call the ```__init__```. Object should not be initialized if its not a correct one.